chore: Added Scarf Tag For Tracking (#246)
litmuschaos/litmus-docs: website/versioned_docs/version-1.2.0/admin-mode.md

---
id: version-1.2.0-admin-mode
title: Administrator Mode
sidebar_label: Administrator Mode
original_id: admin-mode
---
------

###  What is Adminstator Mode? 

> Admin Mode is a priviledged mode that LitmusChaos support in order to make SRE's/ Cluster Admins/ Developers life easier.

Reviewer: ```suggestion
Admin mode is one of the ways the chaos orchestration is set up in Litmus, wherein all chaos resources (i.e., install time resources like the operator, chaosexperiment CRs, chaosServiceAccount/rbac and runtime resources like chaosengine, chaos-runner & experiment jobs) are setup in a single admin namespace (typically, litmus). In other words, centralized administration of chaos. 

This feature is aimed at making the SRE/Cluster Admins life easier by doing away with setting up chaos pre-requisites on a per namespace basis (which may be more relevant in an autonomous/self-service cluster sharing model in dev environments). 
```

> This mode helps to induce chaos in different namespaces, by managing all the chaos resources in a single namespace.

Reviewer: ```suggestion
This mode typically needs a "wider" & "stronger" ClusterRole, albeit one that is still just a superset of the individual experiment permissions. In this mode, the applications in their respective namespaces are subjected to chaos while the chaos job runs elsewhere, i.e., admin namespace. 
```

### How to use Adminstator Mode?

> In order to use Admin Mode, you just have to create a ServiceAccount in Chaos Resources Namespace, that has the permissions to perform operations on different kubernetes resources, in different namespaces.

Reviewer: ```suggestion
In order to use Admin Mode, you just have to create a ServiceAccount in the *admin* or so called *chaos* namespace (`litmus` itself can be used), which is tied to a ClusterRole that has the permissions to perform operations on Kubernetes resources involved in the selected experiments across namespaces.
```

> Just use this ServiceAccount Name in ChaosEngine chaosServiceAccount, this will let you to have all chaos resources in a single namespace, with chaos being performed on various namespaces.

Reviewer: ```suggestion
Use provide this ServiceAccount in ChaosEngine's `.spec.chaosServiceAccount`. 
```

### Example

#### Prepare RBAC Manifest 

```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: litmus
  namespace: litmus
  labels:
    name: litmus
---
# Source: openebs/templates/clusterrole.yaml
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRole
metadata:
  name: litmus
  labels:
    name: litmus
rules:

> - apiGroups: ["*"]

Reviewer: Can we recommend superset of all experiments -- `*/*/*` contains just too much beyond what we need.

kind: ChaosEngine
metadata:
  name: nginx-chaos
  namespace: litmus #Chaos Resources Namespace
spec:
  appinfo:
    appns: 'default' #Application Namespace
    applabel: 'app=nginx'
    appkind: 'deployment'
  # It can be true/false
  annotationCheck: 'true'
  # It can be active/stop
  engineState: 'active'
  #ex. values: ns1:name=percona,ns2:run=nginx
  auxiliaryAppInfo: ''
  chaosServiceAccount: litmus
  monitoring: false
  # It can be delete/retain
  jobCleanUpPolicy: 'delete'
  experiments:
    - name: pod-delete
      spec:
        components:
          env:
            # set chaos duration (in sec) as desired
            - name: TOTAL_CHAOS_DURATION
              value: '30'

            # set chaos interval (in sec) as desired
            - name: CHAOS_INTERVAL
              value: '10'
              
            # pod failures without '--force' & default terminationGracePeriodSeconds
            - name: FORCE
              value: 'false'
```

### Create the ChaosEngine Resource

- Create the ChaosEngine manifest prepared in the previous step to trigger the Chaos.

  `kubectl apply -f chaosengine.yml`

### Watch Chaos Engine 

- Describe Chaos Engine for chaos steps.

>   `kubectl describe chaosengine <chaos-engine> -n <chaos-resources-namespace>`

Reviewer: ```suggestion
  `kubectl describe chaosengine nginx-chaos -n litmus`
```

### Watch Chaos progress

- View pod terminations & recovery by setting up a watch on the pods in the application namespace

>   `watch -n 1 kubectl get pods -n <application-namespace>`

Reviewer: ```suggestion
  `watch -n 1 kubectl get pods -n default`
```

### Check Chaos Experiment Result

- Check whether the application is resilient to the pod failure, once the experiment (job) is completed. The ChaosResult resource name is derived like this: `<ChaosEngine-Name>-<ChaosExperiment-Name>`.

>   `kubectl describe chaosresult nginx-chaos-pod-delete -n <application-namespace>`

Reviewer: ```suggestion
  `kubectl describe chaosresult nginx-chaos-pod-delete -n litmus`
```

This feature is aimed at making the SRE/Cluster Admins life easier by doing away with setting up chaos pre-requisites on a per namespace basis (which may be more relevant in an autonomous/self-service cluster sharing model in dev environments).
This mode typically needs a "wider" & "stronger" ClusterRole, albeit one that is still just a superset of the individual experiment permissions. In this mode, the applications in their respective namespaces are subjected to chaos while the chaos job runs elsewhere, i.e., admin namespace. 

### How to use Adminstator Mode?

In order to use Admin Mode, you just have to create a ServiceAccount in the *admin* or so called *chaos* namespace (`litmus` itself can be used), which is tied to a ClusterRole that has the permissions to perform operations on Kubernetes resources involved in the selected experiments across namespaces.
Provide this ServiceAccount in ChaosEngine's .spec.chaosServiceAccount.

### Example

#### Prepare RBAC Manifest 

Here is an RBAC definition, which in essence is a superset of individual experiments RBAC that has the permissions to run all chaos experiments across different namespaces.

> [embedmd]:# (https://raw.githubusercontent.com/litmuschaos/chaos-charts/master/charts/generic/container-kill/rbac.yaml yaml)

Reviewer 2: Can you please use a correct link: https://raw.githubusercontent.com/litmuschaos/pages/master/docs/litmus-admin-rbac.yaml

This feature is aimed at making the SRE/Cluster Admins life easier by doing away with setting up chaos pre-requisites on a per namespace basis (which may be more relevant in an autonomous/self-service cluster sharing model in dev environments).
This mode typically needs a "wider" & "stronger" ClusterRole, albeit one that is still just a superset of the individual experiment permissions. In this mode, the applications in their respective namespaces are subjected to chaos while the chaos job runs elsewhere, i.e., admin namespace. 

### How to use Adminstator Mode?

In order to use Admin Mode, you just have to create a ServiceAccount in the *admin* or so called *chaos* namespace (`litmus` itself can be used), which is tied to a ClusterRole that has the permissions to perform operations on Kubernetes resources involved in the selected experiments across namespaces.

> Use provide this ServiceAccount in ChaosEngine's `.spec.chaosServiceAccount`. 

Reviewer 2: Provide this ServiceAccount in ChaosEngine's `.spec.chaosServiceAccount`.

### Example

#### Prepare RBAC Manifest 

> https://litmuschaos.github.io/pages/litmus-admin-rbac.yaml

Reviewer 2: Can we make this section more descriptive or use embedmd?

Reviewer 3: Can you help me with embedmd? I dont have knowledge around it
